### DISCUSSION TITLE
ui.jln.dev for `shadcn-theme-editor`

### DISCUSSION THREAD

**Discussion by programming-with-ia** [jln13x/ui.jln.dev#27] [Show and tell]
@jln13x  I’m currently using the following script to fetch theme data:
```
async function fetchThemeById(id: string) {
  const inputJson = JSON.stringify({ "0": { "json": { "id": id } } });
  const encodedInput = encodeURIComponent(inputJson);
  const url = `https://ui.jln.dev/api/trpc/theme.byId?batch=1&input=${encodedInput}`;

  try {
    const response = await fetch(url);
    
    if (!response.ok) {
      throw new Error(`Network response was not ok: ${response.statusText}`);
    }

    const data:any = await response.json();
    console.log(JSON.stringify(data[0].result.data.json.name));
    return data;

  } catch (error) {
    console.error('Fetch error:', error);
  }
}

// Example usage
const newId = 'UvQq6Ww4TeiwkEKg';
fetchThemeById(newId);

```
for [shadcn-theme-editor](https://github.com/programming-with-ia/shadcn-theme-editor/)

### how it works
Users can copy a shareable link from [ui.jln.dev](https://ui.jln.dev/) and paste it into the shadcn-theme-editor. This allows users to further customize the theme in their own projects.

If anyone has a better method for fetching themes, please let me know. I haven’t implemented this feature yet.

**Comment by jln13x** [comment]
i dont plan on adding proper documented external api, so feel free to use it the way you are doing rn. 
this api might change in the future tho so dont fully rely on it

**Reply by programming-with-ia** [reply]
thanks 
i update it when api change in the future
